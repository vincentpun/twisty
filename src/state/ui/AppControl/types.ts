export type AppControlSelection = string | AppControlSelectionSection;

export enum AppControlSelectionSection {
  StartingLocation = 'startingLocation',
  DropoffPoints = 'dropoffPoints',
}
