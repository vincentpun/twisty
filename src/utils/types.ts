export type RecursivePartial<T> = {
  [P in keyof T]?: T[P] extends Array<infer U>
    ? Array<RecursivePartial<U>>
    : T[P] extends ReadonlyArray<infer U>
      ? ReadonlyArray<RecursivePartial<U>>
      : RecursivePartial<T[P]>
};
