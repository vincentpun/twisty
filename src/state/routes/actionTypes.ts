export enum RoutesActionsTypes {
  CreateQuery = 'ROUTES_QUERY_CREATE',
  FetchRoutesToken = 'ROUTES_QUERY_FETCH_ROUTES',
  FetchRoutesTokenReceive = 'ROUTES_QUERY_FETCH_ROUTES_TOKEN_RECEIVE',
  PollResult = 'ROUTES_QUERY_POLL_RESULT',
  PollResultReceiveInProgress = 'ROUTES_QUERY_POLL_RESULT_RECEIVE_IN_PROGRESS',
  QuerySuccess = 'ROUTES_QUERY_SUCCESS',
  QueryReceive = 'ROUTES_QUERY_RECEIVE',
  QueryError = 'ROUTES_QUERY_ERROR',
}
