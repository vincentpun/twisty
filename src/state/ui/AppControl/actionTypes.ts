export enum AppControlUIActionTypes {
  Select = 'APP_CONTROL_SELECT',
}
