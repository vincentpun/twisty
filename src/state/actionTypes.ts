import { UIActionTypes } from './ui/actionTypes';
import { RoutesActionsTypes } from './routes/actionTypes';

export type ActionTypes = UIActionTypes | RoutesActionsTypes;
