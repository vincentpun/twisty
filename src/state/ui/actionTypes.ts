import { AppControlUIActionTypes } from './AppControl/actionTypes';

export enum AppUIActionTypes {
  SwitchMode = 'APP_UI_SWITCH_MODE',
}

export type UIActionTypes = AppUIActionTypes | AppControlUIActionTypes;
