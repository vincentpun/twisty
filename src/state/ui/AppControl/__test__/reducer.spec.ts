import appControl, * as reducer from '../reducer';
import { AppControlUIActionTypes } from '../actionTypes';
import { AppControlSelection, AppControlSelectionSection } from '../types';

describe('appControl', () => {
  it('matches the snapshot', () => {
    const state = appControl(undefined, { type: null });
    expect(state).toMatchSnapshot();
  });

  describe('selection', () => {
    it('matches the snapshot', () => {
      const state = reducer.selection(undefined, { type: null, payload: null });
      expect(state).toMatchSnapshot();
    });

    it('handles AppControlUIActionTypes.Select', () => {
      let state = reducer.selection(AppControlSelectionSection.StartingLocation,{
        type: AppControlUIActionTypes.Select,
        payload: null,
      });
      expect(state).toEqual(null);

      state = reducer.selection(null, {
        type: AppControlUIActionTypes.Select,
        payload: AppControlSelectionSection.StartingLocation,
      });
      expect(state).toEqual(AppControlSelectionSection.StartingLocation);

      state = reducer.selection(null, {
        type: AppControlUIActionTypes.Select,
        payload: 'test',
      });
      expect(state).toEqual('test');
    });
  });
});
