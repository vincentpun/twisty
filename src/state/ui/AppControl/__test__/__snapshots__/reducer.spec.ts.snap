// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appControl matches the snapshot 1`] = `
Object {
  "selection": null,
}
`;

exports[`appControl selection matches the snapshot 1`] = `null`;
